### DISCUSSION TITLE
Another Mac M1 Error/Workaround

### DISCUSSION THREAD

**Discussion by JohnEMagee** [smartcontractkit/full-blockchain-solidity-course-py#1233] [Q&A] [answered]
WOrking on FundMe, I of course ran into an issues as the default chinlink imports do not work because they are compiled in 0.6.0 and we know Mac M1 can't compile 0.6.0 so I'm working in 0.8.12.

My depenences are now based on version 0.4.0 of the chainlink-brownie-contracts and I found the most recent versions of the ETH / USD converter code:
`import "@chainlink/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol";`
However, making a similar change to the SafeMatchChainLink doesn't work
`import "@chainlink/contracts/src/v0.8/vendor/SafeMathChainlink.sol` yields an error 

`ParserError: Source "/Users/johnmagee/.brownie/packages/smartcontractkit/chainlink-brownie-contracts@0.4.0/contracts/src/v0.8/vendor/SafeMathChainlink.sol" not found: File not found. Searched the following locations`

Is there a current version of SafeMath that works with sol 0.8.12 - if so how do i find it properly - if not please let me know as I will have to alter the code to work around whatever isn't working.

Thank you

**Comment by cromewar** [comment, marked as answer]
Hello @JohnEMagee 
the reason is because since version 0.8.0 of solidity you don't need `SafeMath` at all, so that library is not update for recent solidity version, you can just delete that import and also the `using` statement.

**Reply by JohnEMagee** [reply]
Thank you

Now that I'm home and studying this part worked, but I had to also do the payable thing.

One more question - it's not an error but a warning

`Warning: Visibility for constructor is ignored. If you want the contract to be non-deployable, making it "abstract" is sufficient.
  --> contracts/FundMe.sol:16:5:`

Can you advise what this means?
